sapi/cli/php_cli.c

```diff
@@ -1208,6 +1212,11 @@ int main(int argc, char *argv[])
 	 */
 	argv = save_ps_args(argc, argv);
 
+#ifdef PHP_WIN32
```

Comment:
This also needs an `#ifndef PHP_CLI_WIN32_NO_CONSOLE` guard.

Maybe it is worth incorporating a few lines down in the `#ifdef PHP_WIN32` block that call setmode()? However to keep things consistent you would need to replace the `_fileno(std...)` calls with the `STD..._FILENO` defines.